src/crimson/osd/pg_backend.cc

```diff
@@ -868,14 +883,25 @@ seastar::future<> PGBackend::omap_get_vals(
       encode(num, osd_op.outdata);
       osd_op.outdata.claim_append(result);
       encode(truncated, osd_op.outdata);
-      return seastar::now();
-    });
+      return ll_read_errorator::now();
+    }).handle_error(
+      crimson::ct_error::enodata::handle([&osd_op] {
+        uint32_t num = 0;
+        bool truncated = false;
+        encode(num, osd_op.outdata);
```

Comment:
Hmm, `encode()` should be taking by `const` reference, so a temporary should be able to bind to it. Maybe:

```suggestion
        encode(uint32_t { 0} /* num */, osd_op.outdata);
        encode(bool { false } /* truncated */, osd_op.outdata);
```
```

Reply:
thanks. temporary objects also work. one big step closer to one liner!

but seriously, i'd prefer using named arguments. but changed anyway.